### PR TITLE
(maint) Add puppetfile-resolver gem to Bolt projects

### PR DESCRIPTION
**Pull request by beechtom**
Bolt now depends on the `puppetfile-resolver` gem as of
puppetlabs/bolt#1656. This adds the gem and its dependency `molinillo`
to the `bolt-runtime` and `pe-bolt-server-runtime`.

### CODE REVIEW

**Comment by beechtom** [comment]
Tom Beech  8 minutes ago
! mirrorsource https://rubygems.org/downloads/puppetfile-resolver-0.1.0.gem

kerminatorAPP  8 minutes ago
@tombeech: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/puppetfile-resolver-0.1.0.gem md5: 5fcb2ee3525feae3863f854795a64079

Tom Beech  7 minutes ago
! mirrorsource https://rubygems.org/downloads/molinillo-0.6.6.gem

kerminatorAPP  7 minutes ago
@tombeech: Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/molinillo-0.6.6.gem md5: f0f9d1fbb7277e3e4ae6c4be81fc0a92

File: configs/projects/_shared-pe-bolt-server.rb [hunk 1]
```diff
@@ -105,6 +105,7 @@
 proj.component('rubygem-log4r')
 proj.component('rubygem-logging')
 proj.component('rubygem-minitar')
+proj.component('rubygem-molinillo')
```
Comment (donoghuc) [inline configs/projects/_shared-pe-bolt-server.rb]: Can you file a RE ticket with a new library request: 
image

Comment (donoghuc) [reply configs/projects/_shared-pe-bolt-server.rb]: The external gem i'm sure is fine (MIT licence) but I think all "new" gems shipped with PE will need to go through this process (it is very quick).

**Comment by donoghuc** [comment]
You dont have to do this, but i've generally taken an opportunity to update all the gems when I add a new one. I know there is a https://rubygems.org/gems/bcrypt_pbkdf/versions/1.0.1 that got missed bc the Gemfile.lock last used to feed the `generate.rb` script was missing those non-bolt.gemspec deps.

File: configs/projects/_shared-pe-bolt-server.rb (hunk 1 above)
Comment (beechtom) [reply configs/projects/_shared-pe-bolt-server.rb]: Good to know. 
molinillo: https://tickets.puppetlabs.com/browse/RE-13299
puppetfile-resolver: https://tickets.puppetlabs.com/browse/RE-13300

**Comment by beechtom** [comment]
Tom Beech  12 minutes ago
!mirrorsource https://rubygems.org/downloads/aws-partitions-1.280.0.gem

kerminatorAPP  12 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/aws-partitions-1.280.0.gem md5: 7fa7f6facabbc055aabf76e556a9555a

Tom Beech  12 minutes ago
!mirrorsource https://rubygems.org/downloads/aws-sdk-ec2-1.147.0.gem

kerminatorAPP  12 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/aws-sdk-ec2-1.147.0.gem md5: 8d692c65b937bbd1d469fd92667bcacd

Tom Beech  12 minutes ago
!mirrorsource https://rubygems.org/downloads/aws-sigv4-1.1.1.gem

kerminatorAPP  12 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/aws-sigv4-1.1.1.gem md5: 516d8371db693d375dcdc57e862aa930

Tom Beech  12 minutes ago
!mirrorsource https://rubygems.org/downloads/bindata-2.4.6.gem

kerminatorAPP  12 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/bindata-2.4.6.gem md5: c68f27444ca1d17b292889d67c3d92c0

Tom Beech  12 minutes ago
!mirrorsource https://rubygems.org/downloads/concurrent-ruby-1.1.6.gem

kerminatorAPP  12 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/concurrent-ruby-1.1.6.gem md5: 59c0844fecefd0b051ff0db64577be58

Tom Beech  11 minutes ago
!mirrorsource https://rubygems.org/downloads/faraday-0.17.3.gem

kerminatorAPP  11 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/faraday-0.17.3.gem md5: aa0eb149651aa6185ca15b31edafb156

Tom Beech  10 minutes ago
!mirrorsource https://rubygems.org/downloads/faraday_middleware-0.13.1.gem

kerminatorAPP  10 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/faraday_middleware-0.13.1.gem md5: 58f99e8eaf49d9413f5cb5853c41c567

Tom Beech  10 minutes ago
!mirrorsource https://rubygems.org/downloads/ffi-1.12.2.gem

kerminatorAPP  10 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/ffi-1.12.2.gem md5: 28dc3d1294a04b728d24ba025e331b13

Tom Beech  10 minutes ago
!mirrorsource https://rubygems.org/downloads/gettext-setup-0.34.gem

kerminatorAPP  10 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/gettext-setup-0.34.gem md5: 72e511431d138a7c3e062d5e06989a40

Tom Beech  10 minutes ago
!mirrorsource https://rubygems.org/downloads/locale-2.1.3.gem

kerminatorAPP  10 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/locale-2.1.3.gem md5: f5bef9eed8e8c40417a3ab68fa34f477

Tom Beech  9 minutes ago
!mirrorsource https://rubygems.org/downloads/multi_json-1.14.1.gem

kerminatorAPP  9 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/multi_json-1.14.1.gem md5: ff088e41a3af364202670f3afdead842

Tom Beech  9 minutes ago
!mirrorsource https://rubygems.org/downloads/puppet-resource_api-1.8.12.gem

kerminatorAPP  9 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/puppet-resource_api-1.8.12.gem md5: 3e4412252e4574bbf86dd9e85a318f22

Tom Beech  9 minutes ago
!mirrorsource https://rubygems.org/downloads/puppet-strings-2.4.0.gem

kerminatorAPP  9 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/puppet-strings-2.4.0.gem md5: 62aa815177bd9a1c3808faccbb5ed1b8

Tom Beech  9 minutes ago
!mirrorsource https://rubygems.org/downloads/puppet_forge-2.3.3.gem

kerminatorAPP  9 minutes ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/puppet_forge-2.3.3.gem md5: 1bd29a07d2b7531d32c1dccd2bb26623

Tom Beech  1 minute ago
!mirrorsource https://rubygems.org/downloads/bcrypt_pbkdf-1.0.1.gem

kerminatorAPP  1 minute ago
Success! https://artifactory.delivery.puppetlabs.net/artifactory/generic__buildsources/buildsources/bcrypt_pbkdf-1.0.1.gem md5: 6de346254ec38dd5408e5feeb11b9dd8

File: configs/components/rubygem-ffi.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 component "rubygem-ffi" do |pkg, settings, platform|
-  pkg.version '1.9.25'
-  pkg.md5sum "e8923807b970643d9e356a65038769ac"
+  pkg.version '1.12.2'
```
Comment (donoghuc) [inline configs/components/rubygem-ffi.rb]: Any gem that is shared by multiple projects needs special treatment. ffi in particular is sketchy for bolt bc it was not compiling on osx. BUT i think maybe we removed the OSX that it was failing for? There are some other gems you will want to grep /projects for (I think like multi_json). 
```
cas@cas-ThinkPad-T460p:~/working_dir/puppet-runtime/configs/projects$ ag multi_json
agent-runtime-5.5.x.rb
40:  proj.component 'rubygem-multi_json'

agent-runtime-master.rb
32:  proj.component 'rubygem-multi_json'

pe-installer-runtime.rb
134:  proj.component 'rubygem-multi_json'

agent-runtime-6.4.x.rb
30:  proj.component 'rubygem-multi_json'

bolt-runtime.rb
170:  proj.component 'rubygem-multi_json'
```

Comment (donoghuc) [reply configs/components/rubygem-ffi.rb]: Basically if there is a gem that is shared by other runtimes (especially puppet agent) we need to either ask those maintainers if it is OK to bump or we need to manage the gem with a version like this pattern: 
https://github.com/puppetlabs/puppet-runtime/blob/41063e16d7bfb9add8cefcfd1d159cbbc792c9c1/configs/projects/bolt-runtime.rb#L10
https://github.com/puppetlabs/puppet-runtime/blob/41063e16d7bfb9add8cefcfd1d159cbbc792c9c1/configs/components/rubygem-deep_merge.rb#L3-L13

**Comment by beechtom** [comment]
Pulled the commit updating the gems. I'll raise a separate PR for that so it doesn't potentially hold up Bolt-specific work.

**Comment by beechtom** [comment]
https://tickets.puppetlabs.com/browse/RE-13299

```
Molinillo is approved for use!
```

**Comment by beechtom** [comment]
https://tickets.puppetlabs.com/browse/RE-13300

```
puppetfile-resolver is approved for use!
```

**Pull request merged**